File: Leetcode/cpp/sol242.cpp
//https://leetcode.com/problems/valid-anagram/description/


class Solution {
public:

    bool isAnagram(string s, string t) {
        if(s.size() != t.size()){
            return false;
        }
        vector<int> index(26, 0);

        for(int i = 0; i < s.size(); i++){
            int idx1 = s[i] - 'a';
            index[idx1]++;
            int idx2 = t[i] - 'a';
            index[idx2]--;
        }

        for(auto i:index){
            if(i != 0){
                return false;
            }
        }
        return true;
    }
};

File: Leetcode/typescript/sol242.ts

// https://leetcode.com/problems/valid-anagram/

const counter = (word:string) => {
    const obj = {};
    for(let i of word){
        if(!(i in obj)){
            obj[i] = 0;
        }
        obj[i]++;
    }
    return obj;
}

function isAnagram(s: string, t: string): boolean {
    if(s.length !== t.length){
        return false;
    }

    const s_obj = counter(s);
    const t_obj = counter(t);

    for(let ch in s_obj){
        if(s_obj[ch] !== t_obj[ch]){
            return false;
        }
    }

    return true;

    // if(s.length !== t.length){
    //     return false;
    // }
    // return s.split("").sort().join() === t.split("").sort().join()
};